src/main/java/at/ac/tuwien/kr/alpha/Alpha.java

```diff
@@ -71,11 +80,36 @@ public Program readProgram(InputConfig cfg) throws IOException {
 			retVal.accumulate(this.readProgramFiles(cfg.isLiterate(), cfg.getPredicateMethods(), cfg.getFiles()));
 		}
 		if (!cfg.getAspStrings().isEmpty()) {
-			retVal.accumulate(this.readProgramString(StringUtils.join(cfg.getAspStrings(), System.lineSeparator()), cfg.getPredicateMethods()));
+			retVal.accumulate(this.readProgramString(StringUtils.join(cfg.getAspStrings(), "\n"), cfg.getPredicateMethods()));
+		}
+		retVal = this.doProgramTransformations(retVal);
+		if (cfg.isWriteDependencyGraph()) {
+			// FIXME ProgramAnalysis is again created in grounder, should generally do this
+			// in Alpha and pass into grounder (not too critical, only done here in case
+			// user requests dot output for dependency graph)
+			ProgramAnalysis analysis = new ProgramAnalysis(retVal);
+			new DependencyGraphWriter().writeAsDot(analysis.getDependencyGraph(), cfg.getDepGraphTarget(), this.config.isDebugInternalChecks());
 		}
 		return retVal;
 	}
 
+	private Program doProgramTransformations(Program program) {
```

Comment:
I do not fully understand why this code is now duplicated (here and in `NaiveGrounder`)? It seems to be an intermediate version of #185 where program transformations are partially refactored. Can you remove from this PR the partial refactoring of the program transformations?